### ISSUE TITLE
README: requirements and installation instructions

### ISSUE THREAD

**Issue by HaleTom** [luk1337/ih8sn#9]
I came here via the Android Police article that mentioned the project.

It said custom ROMs only.  I'm running a tired MIUI 12.5.  Will it work on rooted stock ROMS?

Perhaps you could answer generally for everyone via a README with requirements and installation instructions?

**Comment by luk1337** [comment]
If you're rooted, then I'd stick with Magisk modules. They should be able to do way more powerful workarounds out of the box. This tool is mostly intended for people on custom ROMs who don't want to use Magisk.

**Comment by walternewtz** [comment]
> If you're rooted, then I'd stick with Magisk modules. They should be able to do way more powerful workarounds out of the box. This tool is mostly intended for people on custom ROMs users who don't want to use Magisk.

How to use it in lineage os rom without magisk installed??

**Comment by luk1337** [comment]
@walternewtz you can get some hints by reading scripts in repo.

**Comment by MajorMahadevan** [comment]
@luk1337 sorry to bother you. But idk how to install this zip. I tried using a custom recovery. ( Kinda noob ).

**Comment by fraction1** [comment]
@luk1337 can you please explain how to use / install the zip?
I hate safetynet 2. Running the latest lineageOs on my oneplus6.  (No magisk) thanx in advance

**Comment by CitizenDroid** [comment]
> If you're rooted, then I'd stick with Magisk modules. They should be able to do way more powerful workarounds out of the box. This tool is mostly intended for people on custom ROMs users who don't want to use Magisk.

I'm sure sure your getting tired of being asked this but some instructions would really be helpful here as looking at this thread most are clueless including me and sticking with Magisk is not really an option! as most of us who end up here is because Magisk has stopped supporting MagiskHide and alternatives don't work like Universal SafetyNet Fix and Mishaal Rahman recommends not using the Shamiko module in its current state. so not many options left to us really. I have rooted phone and magisk installed and is not a issue uninstalling butr need tutorial for yours and I'm using ALEXNDR custom rom would your alternative work for my rom!. 
thanks for any assistance you can give any kind of help here would be really appreciated.

**Comment by willheisenberg** [comment]
I really agree i would be very glad as well about some instructions.

**Comment by maneo22** [comment]
I think unfortunately that they do not want to give some installation instructions !

**Comment by uaktags** [comment]
> @walternewtz you can get some hints by reading scripts in repo.

It was stated to look at the scripts in the repo. What part of that seems complicated?

Here's a hint, take a look at the script in the repo. Perhaps push.sh may shed a little light?

**Comment by maneo22** [comment]
Could someone write some clear installation instructions because I don't underdstand what is a repo.

**Comment by uaktags** [comment]
> Could someone write some clear installation instructions because I don't underdstand what is a repo.

Step 1. Configure the conf
Step 2. Read the Push.sh
Step 3. Decide to either Execute Push.sh or run the commands yourself
Step 4. Reboot and profit...or maybe don't...take a chance or wait until something changes with the project to where instructions are given

**Comment by MajorMahadevan** [comment]
> > Could someone write some clear installation instructions because I don't underdstand what is a repo.
> 
> Step 1. Configure the conf Step 2. Read the Push.sh Step 3. Decide to either Execute Push.sh or run the commands yourself Step 4. Reboot and profit...or maybe don't...take a chance or wait until something changes with the project to where instructions are given

Im really sorry sir, but i don't think you understand the situation correctly. I can't understand a word you posted above. And im really sorry to bother you. I have only basic knowledge of flashing a custom rom and following step by step process for the hurdles ahead. So please and please dumb it down so that i can understand. Again im so sorry, so sorry sir.

**Comment by MajorMahadevan** [comment]
> > > Could someone write some clear installation instructions because I don't underdstand what is a repo.
> > 
> > 
> > Step 1. Configure the conf Step 2. Read the Push.sh Step 3. Decide to either Execute Push.sh or run the commands yourself Step 4. Reboot and profit...or maybe don't...take a chance or wait until something changes with the project to where instructions are given
> 
> Im really sorry sir, but i don't think you understand the situation correctly. I can't understand a word you posted above. And im really sorry to bother you. I have only basic knowledge of flashing a custom rom and following step by step process for the hurdles ahead. So please and please dumb it down so that i can understand. Again im so sorry, so sorry sir.

I don't know how to configure the conf.

**Comment by willheisenberg** [comment]
> > Could someone write some clear installation instructions because I don't underdstand what is a repo.
> 
> Step 1. Configure the conf Step 2. Read the Push.sh Step 3. Decide to either Execute Push.sh or run the commands yourself Step 4. Reboot and profit...or maybe don't...take a chance or wait until something changes with the project to where instructions are given

I tried it and the result was that:

restarting adbd as root
remount succeeded
60-ih8sn.sh: 1 file pushed. 0.1 MB/s (734 bytes in 0.012s)
ih8sn: 1 file pushed. 25.5 MB/s (9866784 bytes in 0.368s)
ih8sn.rc: 1 file pushed. 0.0 MB/s (138 bytes in 0.004s)
push.sh: 12: [[: not found
push.sh: 14: [[: not found
ih8sn.conf: 1 file pushed. 0.0 MB/s (285 bytes in 0.008s)

**Comment by willheisenberg** [comment]
And my config looked like this:

BUILD_FINGERPRINT=google/coral/coral:10/QD1A.190821.007/5831595:user/release-keys
BUILD_DESCRIPTION=coral-user 10 QD1A.190821.007 5831595 release-keys
BUILD_SECURITY_PATCH_DATE=2019-10-05
BUILD_TAGS=release-keys
BUILD_TYPE=user
BUILD_VERSION_RELEASE=10
DEBUGGABLE=0
PRODUCT_NAME=coral

Now somebody what went wrong?

**Comment by MajorMahadevan** [comment]
> And my config for a pixel 4 looked like this:
> 
> BUILD_FINGERPRINT=google/coral/coral:10/QD1A.190821.007/5831595:user/release-keys BUILD_DESCRIPTION=coral-user 10 QD1A.190821.007 5831595 release-keys BUILD_SECURITY_PATCH_DATE=2019-10-05 BUILD_TAGS=release-keys BUILD_TYPE=user BUILD_VERSION_RELEASE=10 DEBUGGABLE=0 PRODUCT_NAME=coral
> 
> Now somebody what went wrong?

How did you get there🤷

**Comment by willheisenberg** [comment]
> > And my config for a pixel 4 looked like this:
> > BUILD_FINGERPRINT=google/coral/coral:10/QD1A.190821.007/5831595:user/release-keys BUILD_DESCRIPTION=coral-user 10 QD1A.190821.007 5831595 release-keys BUILD_SECURITY_PATCH_DATE=2019-10-05 BUILD_TAGS=release-keys BUILD_TYPE=user BUILD_VERSION_RELEASE=10 DEBUGGABLE=0 PRODUCT_NAME=coral
> > Now somebody what went wrong?
> 
> How did you get thereshrug

You download the zip file and open the file with a editor.

**Comment by MajorMahadevan** [comment]
> > > And my config for a pixel 4 looked like this:
> > > BUILD_FINGERPRINT=google/coral/coral:10/QD1A.190821.007/5831595:user/release-keys BUILD_DESCRIPTION=coral-user 10 QD1A.190821.007 5831595 release-keys BUILD_SECURITY_PATCH_DATE=2019-10-05 BUILD_TAGS=release-keys BUILD_TYPE=user BUILD_VERSION_RELEASE=10 DEBUGGABLE=0 PRODUCT_NAME=coral
> > > Now somebody what went wrong?
> > 
> > 
> > How did you get thereshrug
> 
> You download the zip file and open the file with a editor.

What editor 😂
Im sorry. I don't know much about coding.

**Comment by willheisenberg** [comment]
> > > > And my config for a pixel 4 looked like this:
> > > > BUILD_FINGERPRINT=google/coral/coral:10/QD1A.190821.007/5831595:user/release-keys BUILD_DESCRIPTION=coral-user 10 QD1A.190821.007 5831595 release-keys BUILD_SECURITY_PATCH_DATE=2019-10-05 BUILD_TAGS=release-keys BUILD_TYPE=user BUILD_VERSION_RELEASE=10 DEBUGGABLE=0 PRODUCT_NAME=coral
> > > > Now somebody what went wrong?
> > > 
> > > 
> > > How did you get thereshrug
> > 
> > 
> > You download the zip file and open the file with a editor.
> 
> What editor joy Im sorry. I don't know much about coding.

Its not about coding here. Dont put it too high! Its a installation and to open the config file is just like opening a text file.

**Comment by uaktags** [comment]
As a reminder, this is a developer forum (github is a "development platform"). There is no warranty in any thing here as we're all just end-users and not the developer. You're taking a "risk" testing out something that you don't know anything about or how to properly use.

If you're on windows:
(Prereq: You need ADB installed on your machine and know how to use it)

Download the release file's zip, currently labelled as "latest" and extract it to a folder you can access (possibly your Downloads folder).
Go to that folder, and right click on ih8sn.conf, and click "Open With" and select Notepad
Edit the configuration with your desired results for your device and click save
Now you're ready to push the results. However, the push.sh is for linux, so right click on that file and open with notepad.
Now you can see the following, this is what you're going to type into your command prompt:
```
adb wait-for-device root
adb wait-for-device remount
adb wait-for-device push 60-ih8sn.sh /system/addon.d/
adb wait-for-device push ih8sn /system/bin/
adb wait-for-device push ih8sn.rc /system/etc/init/
adb wait-for-device push ih8sn.conf /system/etc/
```

This calls ``adb root`` which requires ADB Root in the developer options on your phone (not providing details on that)
Then it remounts the system (if you have issues, review the other issues)
It then pushes the files from your local directory to their respective folders.

You're doing some pretty "advance" things and thats why this is not beginner-friendly and no one but yourself is responsible if anything goes wrong.

I'd recommend you NOT do this and instead wait for proper documentation instead of asking AND attempting to do something without proper first-party support.

**Comment by uaktags** [comment]
> 
how did you execute the push.sh? It's saying that it didn't succeed with the IF statement due to the [[. 
https://github.com/luk1337/ih8sn/blob/7ae81cf8233d04509a14bab448f3be043e8fffa8/push.sh#L9-L16 and
https://github.com/luk1337/ih8sn/blob/7ae81cf8233d04509a14bab448f3be043e8fffa8/push.sh#L18 are both optional pieces of the push, so you can remove them and leave just https://github.com/luk1337/ih8sn/blob/7ae81cf8233d04509a14bab448f3be043e8fffa8/push.sh#L17. The others are for instances where you may have multiple types of configs.

**Comment by willheisenberg** [comment]
> > > > Could someone write some clear installation instructions because I don't underdstand what is a repo.
> > > 
> > > 
> > > Step 1. Configure the conf Step 2. Read the Push.sh Step 3. Decide to either Execute Push.sh or run the commands yourself Step 4. Reboot and profit...or maybe don't...take a chance or wait until something changes with the project to where instructions are given
> > 
> > 
> > I tried it and the result was that:
> > restarting adbd as root remount succeeded 60-ih8sn.sh: 1 file pushed. 0.1 MB/s (734 bytes in 0.012s) ih8sn: 1 file pushed. 25.5 MB/s (9866784 bytes in 0.368s) ih8sn.rc: 1 file pushed. 0.0 MB/s (138 bytes in 0.004s) push.sh: 12: [[: not found push.sh: 14: [[: not found ih8sn.conf: 1 file pushed. 0.0 MB/s (285 bytes in 0.008s)
> 
> how did you execute the push.sh? It's saying that it didn't succeed with the IF statement due to the [[. https://github.com/luk1337/ih8sn/blob/7ae81cf8233d04509a14bab448f3be043e8fffa8/push.sh#L9-L16 and https://github.com/luk1337/ih8sn/blob/7ae81cf8233d04509a14bab448f3be043e8fffa8/push.sh#L18 are both optional pieces of the push, so you can remove them and leave just https://github.com/luk1337/ih8sn/blob/7ae81cf8233d04509a14bab448f3be043e8fffa8/push.sh#L17. The others are for instances where you may have multiple types of configs.

I use Ubuntu and i used: sh push.sh
Now i reduced it to that, what you said:

sh push.sh
restarting adbd as root
remount succeeded
60-ih8sn.sh: 1 file pushed. 0.1 MB/s (734 bytes in 0.009s)
ih8sn: 1 file pushed. 30.0 MB/s (9866784 bytes in 0.314s)
ih8sn.rc: 1 file pushed. 0.0 MB/s (138 bytes in 0.008s)
ih8sn.conf: 1 file pushed. 0.1 MB/s (285 bytes in 0.002s)

It seemed it worked but how can is use it?

**Comment by GitDevMasters** [comment]
For end-users who want some more friendly, i'm here to help!
First, I need to say i'm not english speecher, so please forgive me the bad words.

Second, it's not because Magisk end support to "Magisk Hide" feature that means you can't install modules to Hide Root and ByPass SafetyNet Validations. Here is the link for a .zip with a Universal SafetyNet Fix that works on Magisk 24+ versions:

https://github.com/kdrag0n/safetynet-fix
You just need to enable Zygisk (option inside Magisk APK) and restart your phone to use it.

If you need to handle with "more aggressive detections" such as bank applications, I recommend use Shamiko module, that you can find a better tutorial on this link (guy thats speak English, and I'm not).:
https://www.youtube.com/watch?v=CvmE9XR8Je0

My purpouse here is just to give some help, please, sorry for the inconvenience!

**Comment by MisterY** [comment]
> As a reminder, this is a developer forum (github is a "development platform"). 

If we're talking about the "development platform", we may find that it has some standards. It is quite normal for a project to have a ReadMe file (also Contributing, License, and others) with a description of the project, setup instructions for developers as well as users, etc. 
Reading the code (of every project we use on GitHub) does not really answer the semantics.

**Comment by uaktags** [comment]
> > As a reminder, this is a developer forum (github is a "development platform").
> 
> If we're talking about the "development platform", we may find that it has some standards. It is quite normal for a project to have a ReadMe file (also Contributing, License, and others) with a description of the project, setup instructions for developers as well as users, etc. Reading the code (of every project we use on GitHub) does not really answer the semantics.

Do we have any of these "standards" publicly listed as a requirement to use the "development platform"? It may be quite normal for a project to have a ReadMe file, but nowhere is it a required practice or a standard that needs to be met. Someone decided to post code to their repository, and end user with no experience in the code decided they wanted to run it. Funny thing is, the developer themselves has stated "you can get some hints by reading scripts in repo", so what's the point in bringing up a 6month issue that the developer(s) themselves have decided they did not want to do, just to critique someone who is not *the* developer on the "standards" on how to use github lol.

Reading the code really does answer the semantics....as reading the code will give you the in depth understanding of how it works and how to use it. If you can't read the code, then the semantics of the code and overall project will be lost on you, so you probably shouldn't be on github in the first place trying to use code/software that has no ReadMe, documentation, or warranty. But hey, if don't agree with me, refer back to the developer's second response in this issue or take it up with github themselves about the fact this repo isn't following the "standards". Cheers!
